setup.sh

```diff
@@ -24,63 +32,86 @@ if [ ! -e ./.env ]; then
 	read -p "LDAP server to be used (default to bundled openldap) [$value_default]: " new_value
 	LDAP_SERVER=${new_value:-$value_default}
 
-	value_default="DC=kopano,DC=demo"
-	read -p "LDAP search base [$value_default]: " new_value
-	LDAP_SEARCH_BASE=${new_value:-$value_default}
-
-	value_default="CN=readonly,DC=kopano,DC=demo"
-	read -p "LDAP bind user (needs only read permissions) [$value_default]: " new_value
-	LDAP_BIND_DN=${new_value:-$value_default}
-
-	value_default="kopano123"
-	read -p "LDAP server to be used (default bundled openldap) [$value_default]: " new_value
-	LDAP_BIND_PW=${new_value:-$value_default}
+	if [ "$LDAP_SERVER" != "$value_default" ]; then
+		value_default="kopano123"
+		read -p "Password of the admin user [$value_default]: " new_value
+		LDAP_ADMIN_PASSWORD=${new_value:-$value_default}
+
+		value_default="$LDAP_BASE_DN"
+		read -p "LDAP search base [$value_default]: " new_value
+		LDAP_SEARCH_BASE=${new_value:-$value_default}
+
+		value_default="CN=readonly,$LDAP_BASE_DN"
+		read -p "LDAP bind user (needs only read permissions) [$value_default]: " new_value
+		LDAP_BIND_DN=${new_value:-$value_default}
+
+		value_default="kopano123"
+		read -p "LDAP bind password to be used [$value_default]: " new_value
+		LDAP_BIND_PW=${new_value:-$value_default}
+	else
+		LDAP_ADMIN_PASSWORD=$(random_string)
+		LDAP_SEARCH_BASE="$LDAP_BIND_DN"
+		LDAP_BIND_DN="CN=readonly,$LDAP_BASE_DN"
+		LDAP_BIND_PW=$(random_string)
+	fi
+
+	if [ -f /etc/timezone ]; then
+		value_default=$(cat /etc/timezone)
+	elif [ -f /etc/localtime ]; then
+		value_default=$(readlink /etc/localtime|sed -n 's|^.*zoneinfo/||p')
+	fi
+
+	if [ -z "${value_default}" ]; then
+		value_default="Europe/Berlin".
+	fi
 
-	value_default="Europe/Berlin"
 	read -p "Timezone to be used [$value_default]: " new_value
 	TZ=${new_value:-$value_default}
 
-	value_default="postmaster@kopano.demo"
+	value_default="postmaster@$FQDN"
 	read -p "E-Mail Address displayed for the 'postmaster' [$value_default]: " new_value
 	POSTMASTER_ADDRESS=${new_value:-$value_default}
 
 	value_default="db"
 	read -p "Name/Address of Database server (defaults to the bundled one) [$value_default]: " new_value
 	MYSQL_HOST=${new_value:-$value_default}
 
-	value_default="kopano123"
-	read -p "Password for the MySQL root user [$value_default]: " new_value
-	MYSQL_ROOT_PASSWORD=${new_value:-$value_default}
-
-	value_default="kopanoDbUser"
-	read -p "Username to connect to the database [$value_default]: " new_value
-	MYSQL_USER=${new_value:-$value_default}
-
-	value_default="kopanoDbPw"
-	read -p "Password to connect to the database [$value_default]: " new_value
-	MYSQL_PASSWORD=${new_value:-$value_default}
-
-	value_default="kopano"
-	read -p "Datebase to use for Kopano [$value_default]: " new_value
-	MYSQL_DATABASE=${new_value:-$value_default}
-
+	if [ "$MYSQL_HOST" != "$value_default" ]; then
+		value_default="kopano123"
+		read -p "Password for the MySQL root user [$value_default]: " new_value
```

Comment:
here its the same as with the ldap image. root user not needed, we just need a user to query the database (and potentially create tables)